Yoast/javascript
# Fix unwanted space type change

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Fixes the unwanted change in the space type in the function that adds word boundary to a keyphrase.

## Relevant technical choices:

* If this PR is merged in preparations for 14.6, then it is no user-facing.

## Test instructions

This PR can be tested by following these steps:

* NB: According to tests, the original problematic space-type change did not cause any changes in the recognition of keyphrase.
* Lingo instructions: 
  * Check that this branch does not differ from `master` in space types in the following files:
    * `javascript/packages/yoastseo/src/stringProcessing/addWordboundary.js`
    * `javascript/packages/yoastseo/spec/stringProcessing/addWordBoundarySpec.js`
    * `javascript/packages/yoastseo/spec/stringProcessing/createRegexFromArray.js`
    * `javascript/packages/yoastseo/spec/stringProcessing/createWordRegexSpec.js`
  * Build a local version of the plugin (content-analysis app or Docker) and check whether the keyphrase is being recognized when used in a sentence when being preceded and/or followed by a regular space and by a non-break space.
  * Please delete the ghost `premium-configuration` branch when you are done testing, which is only meant to allow the javascript tests pass.
* QA instructions:
  * Use a single-word keyphrase and use it in the text preceded and followed by a regular space and a non break space. No matter the combination the keyphrase should be recognized as normal.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes https://yoast.atlassian.net/browse/LIN-490
